instrumentation/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafka/KafkaHeadersGetter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafka;
+
+import io.opentelemetry.context.propagation.TextMapGetter;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public final class KafkaHeadersGetter implements TextMapGetter<Headers> {
```

Comment:
There's no need to implement a second getter for Kafka records - just use the original `KafkaHeadersGetter` and pass it a `ConsumerRecord` object instead of just `Headers`

Reply:
Nah, the naming there is wrong.
It's not a Headers getter ...

And I need an explicit Headers getter -- since I used it from both, ProducerRecord and ConsumerRecord.

Reply:
Are you doing manual context propagation in Strimzi? I was wondering why are methods like `extractContext``/inject` public, are you perhaps using them inside your project?

Reply:
Oh, and why do you need to extract context from a `ProducerRecord`? Usually you'd just inject span context into it to make sure it gets propagated.

Reply:
> Are you doing manual context propagation in Strimzi? I was wondering why are methods like `extractContext`/`inject` public, are you perhaps using them inside your project?

They were public in that OpenTracing utils class ... I can make them private.

Reply:
> Oh, and why do you need to extract context from a `ProducerRecord`? Usually you'd just inject span context into it to make sure it gets propagated.

Same -- just impled equivalent code from OpenTracing.
I didn't think or bother with it too much ... but yeah, your observation makes sense. :-)

Reply:
Yeah, we'd like to keep the public API as minimal as possible - so if you're using something in your code, we should think about how to expose it, but if not -- make it private/package-private.